javav2/example_code/s3/src/main/java/com/example/s3/express/S3DirectoriesActions.java

```diff
@@ -0,0 +1,551 @@
+                    }
+
+                    throw new CompletionException("Failed to create access key: " + exception.getMessage(), exception);
+                }
+            });
+    }
+
+    /**
+     * Selects an availability zone ID based on the specified AWS region.
+     *
+     * @return A map containing the selected availability zone details, including the zone name, zone ID, region name, and state.
+     */
+    public CompletableFuture<String> selectAvailabilityZoneIdAsync() {
+        // Request available zones
+        DescribeAvailabilityZonesRequest zonesRequest = DescribeAvailabilityZonesRequest.builder()
+            .build();
+
+        return getEc2AsyncClient().describeAvailabilityZones(zonesRequest)
+            .thenCompose(response -> {
+                List<AvailabilityZone> zonesList = response.availabilityZones();
+
+                if (zonesList.isEmpty()) {
+                    logger.info("No availability zones found.");
+                    return CompletableFuture.completedFuture(null); // Return null if no zones are found
+                }
+
+                // Extract zone IDs
+                List<String> zoneIds = zonesList.stream()
+                    .map(AvailabilityZone::zoneId) // Get the zoneId (e.g., "usw2-az1")
+                    .toList();
+
+                // **Prompt user synchronously** and return CompletableFuture
+                return CompletableFuture.supplyAsync(() -> promptUserForZoneSelection(zonesList, zoneIds))
+                    .thenApply(selectedZone -> {
+                        // Return only the selected Zone ID (e.g., "usw2-az1")
+                        return selectedZone.zoneId();
+                    });
+            })
+            .whenComplete((result, exception) -> {
+                if (exception == null) {
+                    if (result != null) {
+                        logger.info("Selected Availability Zone ID: " + result);
+                    } else {
+                        logger.info("No availability zone selected.");
+                    }
+                } else {
+                    Throwable cause = exception.getCause();
+                    if (cause instanceof Ec2Exception) {
+                        throw new CompletionException("EC2 error while selecting availability zone: " + cause.getMessage(), cause);
+                    }
+                    throw new CompletionException("Failed to select availability zone: " + exception.getMessage(), exception);
+                }
+            });
+    }
+
+    /**
+     * Prompts the user to select an availability zone from the given list.
+     *
+     * @param zonesList the list of availability zones
+     * @param zoneIds the list of zone IDs
+     * @return the selected AvailabilityZone
+     */
+    private static AvailabilityZone promptUserForZoneSelection(List<AvailabilityZone> zonesList, List<String> zoneIds) {
```

Comment:
Why not just pass in the list of AZs, since the zoneId is property of an AvailabliltyZone object? Rather than a separate list of zoneIds that was previously extracted from the list of AvailablityZones in the calling method?

Reply:
Lets talk about this one